frontend/tests/playwright/msw/request-handler.ts

```diff
@@ -0,0 +1,233 @@
+ * @see https://mswjs.io/docs/api/context
+ * @see https://mswjs.io/docs/api/response
+ *
+ */
+export type MockedResponseInfo = {
+   status: number;
+   headers?: Record<string, string | string[]>;
+   body?: Record<string, any> | string | number | boolean | null;
+   responseType?: 'json' | 'text' | 'xml' | 'raw';
+};
+
+/**
+ * These are some options that can be passed to configure the request handler.
+ * @property `once` - If true, the handler will only be used once.
+ * @see https://mswjs.io/docs/api/response/once
+ * @property `passthrough` - If true, the request will be bypassed and performed
+ * as-is.
+ * @see https://mswjs.io/docs/api/request/passthrough
+ * @property `customResolver` - If provided, this will be used as the resolver
+ * for the handler. This is useful if you want to do something more complex than
+ * just returning a static response.
+ * @see https://mswjs.io/docs/basics/response-resolver
+ */
+export type HandlerOptions = {
+   once: boolean;
+   passthrough: boolean;
+   customResolver?: GraphQLResolver | RestResolver;
+};
+
+export default class Handler {
+   /**
+    * Creates a new GraphQL handler.
+    */
+   static create({
+      request,
+      response,
+      options,
+   }: {
+      request: Omit<RequestInfo, 'method'> & { method: GraphQLMethods };
+      response: Omit<
+         MockedResponseInfo,
+         'headers' | 'body' | 'responseType'
+      > & {
+         body: Record<string, any>;
+      };
+      options?: HandlerOptions;
+   }): GraphQLHandler;
+
+   /**
+    * Creates a new Rest Handler.
+    */
+   static create({
+      request,
+      response,
+      options,
+   }: {
+      request: Omit<RequestInfo, 'method'> & { method: RestMethods };
+      response: MockedResponseInfo;
+      options?: HandlerOptions;
+   }): RestHandler;
+
+   static create({
+      request,
+      response,
+      options = { once: false, passthrough: false },
+   }: {
+      request: RequestInfo;
+      response: MockedResponseInfo;
+      options?: HandlerOptions;
+   }): RequestHandler {
+      const { endpoint, method } = request;
+      const { status, headers, body, responseType } = response;
+      const { once, passthrough, customResolver } = options;
+
+      if (isGraphQLMethod(method)) {
+         if (!isRecord(body)) {
+            throw new Error(`Invalid GraphQL body ${body}. Must be an object`);
+         }
+
+         return new GraphQLHandler(
+            method,
+            endpoint,
+            /**
+             * This refers to the Path,or location, of the endpoint. By using a
+             * wildcard, we are saying that we want to match any endpoint
+             * regardless of location the request is made from.
+             */
+            '*',
+            customResolver ??
+               ((req, res, ctx) => {
+                  if (passthrough) return req.passthrough();
+                  if (once) return res.once(ctx.status(status), ctx.data(body));
```

Comment:
Since `passthrough` and `once` don't do anything when `customResolver` is present, I might try to make the `HandlerOptions` type reflect that:
`type GraphQLHandlerOptions = { customResolver: GraphQLResolver } | { passthrough?: boolean; once: boolean; }`